src/HandlerStack.php

```diff
@@ -252,6 +252,6 @@ private function debugCallable($fn): string
                 : "callable(['" . \get_class($fn[0]) . "', '{$fn[1]}'])";
         }
 
-        return 'callable(' . \spl_object_hash($fn) . ')';
+        return 'callable(' . \spl_object_hash((object) $fn) . ')';
```

Comment:
What codepath causes this cast to be needed?

Reply:
👍 I had this in mind.. forgot to comment about it..

Reply:
`spl_object_hash` expects an object to create a hash out of it, so my assumption given that this is related to a debug method was that the safest bet here would be to try reaching that with a typecast.

I've analyzed all `Middleware::$stack` writes, and we can be safe its items are covered in `\GuzzleHttp\Tests\HandlerStackTest::testCanPrintMiddleware`. The only usages of `debugMessage` are in `__toString`.

Reply:
@rvanlaak not sure what's your conclusion here? I am still in favor of removing the typecast (even after your comment) since I consider it safe.

https://3v4l.org/EL7ZV

https://3v4l.org/EedLd

check the OP codes as well.

Reply:
Ok As in, replace it?

```
/** @var object $fn */
```

Fixed in 2a5e4d5de1df05c96cb4b4b80ef2fe1be59b27db

Reply:
Hi.. I am afraid this is not sufficient.. since it can be a callback... 

just a note here.. and as a feedback.. note that PHPStan is just a tool.. there is no need to address EVERY baseline error even in the wrong way just in order not to have baseline errors.

same goes for this as well https://github.com/guzzle/guzzle/pull/2677#discussion_r436955245

Reply:
Can array-syntax callbacks actually be cast to an object? I think both variants of the code are broken? Do we need to make a closure from the callable first?

Reply:
array-syntax callbacks get covered in the lines above:

```php
        if (\is_array($fn)) {
            return \is_string($fn[0])
                ? "callable({$fn[0]}::{$fn[1]})"
                : "callable(['" . \get_class($fn[0]) . "', '{$fn[1]}'])";
        }
```